### PR TITLE
Assert eqw instruction

### PR DESCRIPTION
**Pull request by hackaugusto**
## Describe your changes

Adding the instruction `assert_eqw`, [requested here](https://github.com/0xPolygonMiden/miden-base/pull/26#issuecomment-1449196510)

## Checklist before requesting a review
- Repo forked and branch created from `next` according to naming convention.
- Commit messages and codestyle follow [conventions](./CONTRIBUTING.md).
- Relevant issues are linked in the PR description.
- Tests added for new functionality.
- Documentation/comments updated according to changes.

### CODE REVIEW

File: assembly/src/assembler/instruction/mod.rs [hunk 1]
```diff
@@ -40,6 +40,8 @@ impl Assembler {
         let result = match instruction {
             Instruction::Assert => span.add_op(Assert),
             Instruction::AssertEq => span.add_ops([Eq, Assert]),
+            Instruction::AssertEqw => span
+                .add_ops([MovUp4, Eq, Assert, MovUp3, Eq, Assert, MovUp2, Eq, Assert, Eq, Assert]),
```
Comment (frisitano) [inline assembly/src/assembler/instruction/mod.rs:44]: Did you consider putting this in `field_ops.rs` as a function? Most of the other "complex" assembly instructions are defined as a function with docs.  Might be something to consider what do you think?

**Review by frisitano — COMMENTED** [review]
Looks good!  Left a comment in-line.

File: assembly/src/assembler/instruction/mod.rs (hunk 1 above)
Comment (hackaugusto) [reply assembly/src/assembler/instruction/mod.rs]: moved code to `field_ops`

File: assembly/src/assembler/instruction/field_ops.rs [hunk 2]
```diff
@@ -7,6 +7,14 @@ use crate::MAX_EXP_BITS;
 /// Field element representing TWO in the base field of the VM.
 const TWO: Felt = Felt::new(2);
 
+// ASSERTIONS
+// ================================================================================================
+
+/// Asserts that the top two words in the stack are equal.
```
Comment (frisitano) [inline assembly/src/assembler/instruction/field_ops.rs:13]: MIght be worth adding the cycle count to this comment as well.

**Review by frisitano — APPROVED** [review]
Would be great if you could add the cycle count to the doc comment.  Apart from that all looks good.

File: assembly/src/assembler/instruction/field_ops.rs (hunk 2 above)
Comment (hackaugusto) [reply assembly/src/assembler/instruction/field_ops.rs:13]: added

**Review by bobbinth — APPROVED** [review]
Looks good! Thank you!

**Pull request merged**